draft-tiruben-reverse-http.md

```diff
@@ -44,18 +44,18 @@ The HTTP protocol has long supported the ability of clients to access origins vi
    - HTTP request proxies (a.k.a. forward proxies)
-   - Transport proxies (e.g., CONNECT, CONNECT-UDP)
-   - IP relays (e.g., CONNECT-IP)
-   - Oblivious HTTP Relays
+   - Transport proxies (e.g., CONNECT (see Section 9.3.6 of {{?RFC9110}}), CONNECT-UDP {{?RFC9298}})
+   - IP relays (e.g., CONNECT-IP {{?I-D.draft-ietf-masque-connect-ip}})
+   - Oblivious HTTP Relays {{?I-D.draft-ietf-ohai-ohttp}}
 * Origin-selected
    - HTTP gateways (a.k.a. reverse proxies)
    - Transport load balancers
 
 Although these intermediaries differ widely in their functionality, they all generally act as a client when connecting to the origin.  Client-selected intermediaries reach the origin based on its hostname or IP address specified in the HTTP request, while origin-selected intermediaries first translate this destination into a "backend address".
 
-One of the main advantages of origin-selected intermediaries is their ability to defend the origin from attacks, especially Denial of Service (DoS) attacks in which the attacker floods the origin server with requests or packets.  To prevent attackers from simply bypassing the intermediary, common practices include keeping the backend address secret and/or instituting firewall rules that only allow packets from the intermediary.  These practices are reasonably effective with origin-selected intermediaries, but they cannot be used with client-selected intermediaries, as those intermediaries do not know the secret backend address, and the origin does not know their "exit" IPs.
+One of the main advantages of origin-selected intermediaries is their ability to defend the origin from attacks, especially Denial of Service (DoS) attacks in which the attacker floods the origin server with requests or packets.  To prevent attackers from simply bypassing the intermediary, common practices include keeping the backend address secret and/or instituting firewall rules that only allow packets from the intermediary.  These practices are reasonably effective with origin-selected intermediaries, but they cannot be used with client-selected intermediaries, as those intermediaries do not know the secret backend address, and the origin does not know their "exit" IP addresses.
 
-This specification defines a protocol for HTTP connections between origins and arbitrary intermediaries that can limit the impact of DoS attacks.  When this protocol is in use, origins have the ability to partition the infrastructure that serves each intermediary so that attacks on the origin's public address, or via one intermediary, do not affect any other intermediaries.
+This specification defines a protocol for HTTP connections between origins and arbitrary intermediaries that can limit the impact of Layer 3 and Layer 4 DDoS attacks.  When this protocol is in use, origins have the ability to partition the infrastructure that serves each intermediary so that attacks on the origin's IP address via one intermediary do not affect any other intermediaries.
```

Comment:
It seems like this change drops the idea that if an attacker takes the public origin IP offline, the origin might remain accessible via various intermediaries.

Reply:
I did not mean to drop the idea. We can modify the text as follows to better clarity:
This specification defines a protocol for HTTP connections between origins and arbitrary intermediaries that can limit the impact of Layer 3 and Layer 4 DDoS attacks. When this protocol is in use, origins have the ability to partition the infrastructure that serves each intermediary. This ensures that attacks targeting the origin's public IP address or attacks via one intermediary will not affect any other intermediaries. By partitioning the infrastructure, the impact of the attacks is contained within the affected intermediary or the origin's public IP address.